### PR TITLE
Prevent CGO and STATIC_OVERRIDES from trampling themselves if evaluated multiple times

### PR DESCRIPTION
**Pull request by dekkagaijin**
/kind bug

```release-note
NONE
```

### CODE REVIEW

**Comment by BenTheElder** [comment]
I think I'm fine with this, but as a general rule our scripts really shouldn't be sourced repeatedly, and we don't _really_ intended support using these as a library externally ...

File: hack/lib/golang.sh [hunk 1]
```diff
@@ -339,25 +339,25 @@ readonly KUBE_COVERAGE_INSTRUMENTED_PACKAGES=(
 # KUBE_CGO_OVERRIDES is a space-separated list of binaries which should be built
 # with CGO enabled, assuming CGO is supported on the target platform.
 # This overrides any entry in KUBE_STATIC_LIBRARIES.
-IFS=" " read -ra KUBE_CGO_OVERRIDES <<< "${KUBE_CGO_OVERRIDES:-}"
-readonly KUBE_CGO_OVERRIDES
+IFS=" " read -ra __KUBE_CGO_OVERRIDES <<< "${KUBE_CGO_OVERRIDES:-}"
+readonly __KUBE_CGO_OVERRIDES
```
Comment (BenTheElder) [inline hack/lib/golang.sh:343]: usually we just have a different name for the user facing one and the internal computed one. I don't think we have any `__KUBE` like this elsewhere.

**Comment by dekkagaijin** [comment]
@BenTheElder FWIW I'm experiencing these statements being executed several times per `make release-in-a-container`, I'd bet that these (and other scripts similarly unable to tolerate multiple `source`ings) have been broken for a while

File: hack/lib/golang.sh (hunk 1 above)
Comment (dekkagaijin) [reply hack/lib/golang.sh]: done

**Comment by dekkagaijin** [comment]
ping :)

**Review by BenTheElder — APPROVED** [review]
lgtm

**Pull request merged**